### PR TITLE
Update ML regression

### PR DESCRIPTION
**Pull request by YinYi000**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [ ] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by zhangqi444 — APPROVED** [review]

File: open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md [hunk 1]
```diff
@@ -34,35 +38,35 @@ In this section, you will learn how to:
...
-
-```{note}
-Click the image above for a video: using Python within VS Code.
+```{seealso}
+Click here for a video : [Setup Python with Visual Studio Code](https://youtu.be/yyQM70vi7V8), by author [Alfredo Deza](https://www.youtube.com/@alfredodeza1).
 ```
 
-1. **Install Python**. Ensure that [Python](https://www.python.org/downloads/) is installed on your computer. You will use Python for many data science and machine learning tasks. Most computer systems already include a Python installation. There are useful [Python Coding Packs](https://code.visualstudio.com/learn/educators/installers?WT.mc_id=academic-77952-leestott) available as well, to ease the setup for some users.
+1 . **Install Python**. Ensure that [Python](https://www.python.org/downloads/) is installed on your computer. You will use Python for many data science and machine learning tasks. Most computer systems already include a Python installation. There are useful [Python Coding Packs](https://code.visualstudio.com/learn/educators/installers?WT.mc_id=academic-77952-leestott) available as well, to ease the setup for some users.
```
[the remark this answers is not in the document]
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md:45]: Yes, it is necessary. if not, the serial number will be wrong

File: open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md [hunk 2]
```diff
@@ -100,9 +110,11 @@ Scikit-learn makes it straightforward to build models and evaluate them for use.
...
 
-> This tutorial was inspired by the [linear regression example](https://scikit-learn.org/stable/auto_examples/linear_model/plot_ols.html#sphx-glr-auto-examples-linear-model-plot-ols-py) on Scikit-learn's web site.
+```{note}
+This tutorial was inspired by the [linear regression example](https://scikit-learn.org/stable/auto_examples/linear_model/plot_ols.html#sphx-glr-auto-examples-linear-model-plot-ols-py) on Scikit-learn's web site.
+```
 
-In the _notebook.ipynb_ file associated to this section, clear out all the cells by pressing the 'trash can' icon.
+In the _regression-tools.ipynb_ file associated to this section, clear out all the cells by pressing the 'trash can' icon.
 
 In this section, you will work with a small dataset about diabetes that is built into Scikit-learn for learning purposes. Imagine that you wanted to test a treatment for diabetic patients. Machine Learning models might help you determine which patients would respond better to the treatment, based on combinations of variables. Even a very basic regression model, when visualized, might show information about variables that would help you organize your theoretical clinical trials.
```
[the remark this answers is not in the document]
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md:119]: There is no extra space before "In this section...."?

File: open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md [hunk 3]
```diff
@@ -148,7 +160,7 @@ This dataset includes the concept of 'sex' as a feature variable important to re
 Now, load up the X and y data.
 
 ```{note}
- Remember, this is supervised learning, and we need a named 'y' target.
+Remember, this is supervised learning, and we need a named 'y' target.
 ```
 
 In a new code cell, load the diabetes dataset by calling `load_diabetes()`. The input `return_X_y=True` signals that `X` will be a data matrix, and `y` will be the regression target.
```
[the remark this answers is not in the document]
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md:166]: There is no extra space before "In a new code cell...."?  where do you mean the space？

File: open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md [hunk 4]
```diff
@@ -219,8 +231,6 @@ Think a bit about what's going on here. A straight line is running through many
 
 Congratulations, you built your first linear regression model, created a prediction with it, and displayed it in a plot!
```
[the remark this answers is not in the document]
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md:232]: no extra space before "Congratulations..."?

File: open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md [hunk 5]
```diff
@@ -219,8 +231,6 @@ Think a bit about what's going on here. A straight line is running through many
...
 
----
-
 ## Self study
 
 In this tutorial, you worked with simple linear regression, rather than univariate or multiple linear regression. Read a little about the differences between these methods, or take a look at [this video](https://www.coursera.org/lecture/quantifying-relationships-regression-models/linear-vs-nonlinear-categorical-variables-ai2Ef)
```
[the remark this answers is not in the document]
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md:236]: no extra space before "In this tutorial..."?

File: open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md [hunk 6]
```diff
@@ -231,7 +241,7 @@ Read more about the concept of regression and think about what kinds of question
 
 Plot a different variable from this dataset. Hint: edit this line: `X = X[:, np.newaxis, 2]`. Given this dataset's target, what are you able to discover about the progression of diabetes as a disease?
```
[the remark this answers is not in the document]
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/regression/tools-of-the-trade.md:242]: no extra space before "Plot a different variable from this dataset..." ?

**Pull request merged**